app/data_model/answer_store.py

```diff
@@ -245,7 +248,24 @@ def upgrade_1_to_2_add_group_instance_id(answer_store, schema):
         answer['group_instance_id'] = get_group_instance_id(schema, answer_store, location, answer['answer_instance'])
 
 
-UPGRADE_TRANSFORMS = (
-    upgrade_0_to_1_update_date_formats,
-    upgrade_1_to_2_add_group_instance_id,
-)
+def upgrade_to_4_remove_empty_answers(answer_store, schema):  # pylint: disable=unused-argument
+    """ Previously, we stored [] and '' as answer values for unanswered, but completed answers.
+
+    This was changed to avoid storing unanswered values entirely, so this upgrade will
+    remove any answers in the store that have 'empty' answer values
+    """
+    answer_store_copy = answer_store.copy()
+
+    for answer in answer_store:
+        if answer['value'] in ('', [], None):
```

Comment:
Why does doc string say `we stored [] and '' as answer values for unanswered,` but this has None as well?